packages/grafana-icons/rollup.config.ts

```diff
@@ -0,0 +1,35 @@
+import resolve from '@rollup/plugin-node-resolve';
```

Comment:
As it is right now it looks like we have a pipeline of: SVG -> TS -> JS + TS definitions.

Do we need rollup and typescript for this package? Could we ditch them both in favour of making this package an ESM only package, and set svgr to output to JS? I don't think there's any of our tooling that can't work with this setup and ESM for Icons will give much better tree-shaking potential than the CJS output. There is the rest spread in `IconBase` but not sure how problematic that would be if we're not transpiling.

As for the types the TS definitions are quite basic with a lot of repetition. Could we add a simple script that generates this too?

```
import React, { SVGProps } from 'react';

type IconSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' | 'xxxl';
interface IconProps extends Omit<SVGProps<SVGSVGElement>, 'onLoad' | 'onError' | 'ref'> {
  size?: IconSize;
  title?: string;
  color?: string;
}

declare const SvgAngleRight: (props: IconProps) => React.JSX.Element;
declare const SvgApps: (props: IconProps) => React.JSX.Element;
declare const SvgBook: (props: IconProps) => React.JSX.Element;
declare const SvgCancel: (props: IconProps) => React.JSX.Element;
export { SvgAngleRight as AngleRight, SvgApps as Apps, SvgBook as Book, SvgCancel as Cancel, type IconProps };

```
I might be missing something but I think in this way the package could be stripped down to only the svgs as src and IconBase.js in the root with the icon components and the index.d.ts generated from a single build command.

Reply:
> Do we need rollup and typescript for this package?

I've been wondering the same and the reasoning for adding it is mainly to use types in IconBase. Also it helps to validate that generated components are properly typed since we've customised the component template. 


> As for the types the TS definitions are quite basic with a lot of repetition. Could we add a simple script that generates this too?

Do you mean to generate index.d.ts that provides types for the generated Icons and IconBase? I can look into that, should be pretty straightforward and simplify the package and CI a lot, altho it does mean losing type safety here. I guess we can always add it back if the component logic becomes too complex 🤔 We can also look into using JSDoc comments to add some type safety without the build step.

Reply:
>altho it does mean losing type safety here

Would we lose type safety? Won't the package.json reference the index.d.ts file which would be pretty much the same as the currently built index.d.ts output?

Reply:
Will it apply when working on the IconBase and other files in the icons package? If yes, then I guess we're all good.

Reply:
> I might be missing something but I think in this way the package could be stripped down to only the svgs as src and IconBase.js in the root with the icon components and the index.d.ts generated from a single build command.

We still the generated icons at least locally as that's where they linked to by lerna.

Reply:
Let's circle back round to this at a later date. I'd rather unblock you so you can iterate on this package.